lib/component.hpp

```diff
@@ -1,24 +1,55 @@
 #ifndef COMPONENT_HPP
 #define COMPONENT_HPP
 
+#include "messages/spa_subscription_reply.hpp"
+#include "messages/spa_subscription_request.hpp"
 #include "spa_communicator.hpp"
+#include "spa_message.hpp"
+#include <iostream>
 #include <memory>
 
 class Component
 {
 public:
   typedef std::shared_ptr<SpaCommunicator> Com;
 
-  Component(Com communicator = nullptr) : communicator(communicator) {}
-  // virtual ~Component(){}
+  Component(Com communicator = nullptr, LogicalAddress address = LogicalAddress(0, 0)) : communicator(communicator),
+                                                                                         address(address),
+                                                                                         dialogId(0) {}
+  virtual ~Component() {}
```

Comment:
Nice, I don't know why this was removed, but is important as we start extending component